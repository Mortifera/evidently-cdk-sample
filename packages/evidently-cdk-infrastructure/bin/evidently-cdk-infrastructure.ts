#!/usr/bin/env node
import 'source-map-support/register';
import * as cdk from 'aws-cdk-lib';
import { EvidentlyCdkInfrastructureStack } from '../lib/evidently-cdk-infrastructure-stack';

export const app = new cdk.App();
new EvidentlyCdkInfrastructureStack(app, 'EvidentlyCdkInfrastructureStack', {
    
});