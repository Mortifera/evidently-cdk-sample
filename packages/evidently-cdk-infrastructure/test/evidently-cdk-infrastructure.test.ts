import { app } from '../bin/evidently-cdk-infrastructure';

const synthedApp = app.synth();

test('Stack Names', () => {
    const stackNames = synthedApp.stacks.map(stack => stack.stackName).sort();
    expect(stackNames).toMatchSnapshot();
});

synthedApp.stacks.forEach(stack => {
    test('Stack Templates - ' + stack.stackName, () => {
        expect(JSON.stringify(stack.template, undefined, 4)).toMatchSnapshot();
    });
});
