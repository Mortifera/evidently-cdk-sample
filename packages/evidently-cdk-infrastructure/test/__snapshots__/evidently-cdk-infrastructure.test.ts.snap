// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack Names 1`] = `
Array [
  "EvidentlyCdkInfrastructureStack",
]
`;

exports[`Stack Templates - EvidentlyCdkInfrastructureStack 1`] = `
"{
    \\"Resources\\": {
        \\"EventsStoreBucket4C2339CD\\": {
            \\"Type\\": \\"AWS::S3::Bucket\\",
            \\"Properties\\": {
                \\"BucketEncryption\\": {
                    \\"ServerSideEncryptionConfiguration\\": [
                        {
                            \\"ServerSideEncryptionByDefault\\": {
                                \\"SSEAlgorithm\\": \\"AES256\\"
                            }
                        }
                    ]
                },
                \\"LifecycleConfiguration\\": {
                    \\"Rules\\": [
                        {
                            \\"ExpirationInDays\\": 30,
                            \\"Status\\": \\"Enabled\\"
                        }
                    ]
                }
            },
            \\"UpdateReplacePolicy\\": \\"Delete\\",
            \\"DeletionPolicy\\": \\"Delete\\"
        },
        \\"EventsStoreBucketPolicyD2BCAE70\\": {
            \\"Type\\": \\"AWS::S3::BucketPolicy\\",
            \\"Properties\\": {
                \\"Bucket\\": {
                    \\"Ref\\": \\"EventsStoreBucket4C2339CD\\"
                },
                \\"PolicyDocument\\": {
                    \\"Statement\\": [
                        {
                            \\"Action\\": [
                                \\"s3:PutObject*\\",
                                \\"s3:GetObject*\\"
                            ],
                            \\"Effect\\": \\"Allow\\",
                            \\"Principal\\": {
                                \\"Service\\": \\"evidently.amazonaws.com\\"
                            },
                            \\"Resource\\": {
                                \\"Fn::Join\\": [
                                    \\"\\",
                                    [
                                        {
                                            \\"Fn::GetAtt\\": [
                                                \\"EventsStoreBucket4C2339CD\\",
                                                \\"Arn\\"
                                            ]
                                        },
                                        \\"/events*\\"
                                    ]
                                ]
                            }
                        },
                        {
                            \\"Action\\": \\"s3:ListBucket\\",
                            \\"Effect\\": \\"Allow\\",
                            \\"Principal\\": {
                                \\"Service\\": \\"evidently.amazonaws.com\\"
                            },
                            \\"Resource\\": {
                                \\"Fn::GetAtt\\": [
                                    \\"EventsStoreBucket4C2339CD\\",
                                    \\"Arn\\"
                                ]
                            }
                        }
                    ],
                    \\"Version\\": \\"2012-10-17\\"
                }
            }
        },
        \\"ProjectProjectCustomResourceCustomResourcePolicy7FFE5362\\": {
            \\"Type\\": \\"AWS::IAM::Policy\\",
            \\"Properties\\": {
                \\"PolicyDocument\\": {
                    \\"Statement\\": [
                        {
                            \\"Action\\": \\"evidently:CreateProject\\",
                            \\"Effect\\": \\"Allow\\",
                            \\"Resource\\": {
                                \\"Fn::Join\\": [
                                    \\"\\",
                                    [
                                        \\"arn:aws:evidently:\\",
                                        {
                                            \\"Ref\\": \\"AWS::Region\\"
                                        },
                                        \\":\\",
                                        {
                                            \\"Ref\\": \\"AWS::AccountId\\"
                                        },
                                        \\":project/test-name\\"
                                    ]
                                ]
                            }
                        },
                        {
                            \\"Action\\": \\"evidently:UpdateProject\\",
                            \\"Effect\\": \\"Allow\\",
                            \\"Resource\\": {
                                \\"Fn::Join\\": [
                                    \\"\\",
                                    [
                                        \\"arn:aws:evidently:\\",
                                        {
                                            \\"Ref\\": \\"AWS::Region\\"
                                        },
                                        \\":\\",
                                        {
                                            \\"Ref\\": \\"AWS::AccountId\\"
                                        },
                                        \\":project/test-name\\"
                                    ]
                                ]
                            }
                        },
                        {
                            \\"Action\\": \\"evidently:DeleteProject\\",
                            \\"Effect\\": \\"Allow\\",
                            \\"Resource\\": {
                                \\"Fn::Join\\": [
                                    \\"\\",
                                    [
                                        \\"arn:aws:evidently:\\",
                                        {
                                            \\"Ref\\": \\"AWS::Region\\"
                                        },
                                        \\":\\",
                                        {
                                            \\"Ref\\": \\"AWS::AccountId\\"
                                        },
                                        \\":project/test-name\\"
                                    ]
                                ]
                            }
                        }
                    ],
                    \\"Version\\": \\"2012-10-17\\"
                },
                \\"PolicyName\\": \\"ProjectProjectCustomResourceCustomResourcePolicy7FFE5362\\",
                \\"Roles\\": [
                    {
                        \\"Ref\\": \\"AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2\\"
                    }
                ]
            }
        },
        \\"ProjectProjectCustomResource4FAFBBAA\\": {
            \\"Type\\": \\"Custom::AWS\\",
            \\"Properties\\": {
                \\"ServiceToken\\": {
                    \\"Fn::GetAtt\\": [
                        \\"AWS679f53fac002430cb0da5b7982bd22872D164C4C\\",
                        \\"Arn\\"
                    ]
                },
                \\"Create\\": {
                    \\"Fn::Join\\": [
                        \\"\\",
                        [
                            \\"{\\\\\\"service\\\\\\":\\\\\\"Evidently\\\\\\",\\\\\\"action\\\\\\":\\\\\\"createProject\\\\\\",\\\\\\"physicalResourceId\\\\\\":{\\\\\\"id\\\\\\":\\\\\\"Project-test-name\\\\\\"},\\\\\\"parameters\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"test-name\\\\\\",\\\\\\"dataDelivery\\\\\\":{\\\\\\"s3Destination\\\\\\":{\\\\\\"bucket\\\\\\":\\\\\\"\\",
                            {
                                \\"Ref\\": \\"EventsStoreBucket4C2339CD\\"
                            },
                            \\"\\\\\\",\\\\\\"prefix\\\\\\":\\\\\\"events\\\\\\"}}}}\\"
                        ]
                    ]
                },
                \\"Update\\": \\"{\\\\\\"service\\\\\\":\\\\\\"Evidently\\\\\\",\\\\\\"action\\\\\\":\\\\\\"updateProject\\\\\\",\\\\\\"physicalResourceId\\\\\\":{\\\\\\"id\\\\\\":\\\\\\"Project-test-name\\\\\\"},\\\\\\"parameters\\\\\\":{}}\\",
                \\"Delete\\": \\"{\\\\\\"service\\\\\\":\\\\\\"Evidently\\\\\\",\\\\\\"action\\\\\\":\\\\\\"deleteProject\\\\\\",\\\\\\"physicalResourceId\\\\\\":{\\\\\\"id\\\\\\":\\\\\\"Project-test-name\\\\\\"},\\\\\\"parameters\\\\\\":{\\\\\\"project\\\\\\":\\\\\\"test-name\\\\\\"}}\\",
                \\"InstallLatestAwsSdk\\": true
            },
            \\"DependsOn\\": [
                \\"ProjectProjectCustomResourceCustomResourcePolicy7FFE5362\\"
            ],
            \\"UpdateReplacePolicy\\": \\"Delete\\",
            \\"DeletionPolicy\\": \\"Delete\\"
        },
        \\"AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2\\": {
            \\"Type\\": \\"AWS::IAM::Role\\",
            \\"Properties\\": {
                \\"AssumeRolePolicyDocument\\": {
                    \\"Statement\\": [
                        {
                            \\"Action\\": \\"sts:AssumeRole\\",
                            \\"Effect\\": \\"Allow\\",
                            \\"Principal\\": {
                                \\"Service\\": \\"lambda.amazonaws.com\\"
                            }
                        }
                    ],
                    \\"Version\\": \\"2012-10-17\\"
                },
                \\"ManagedPolicyArns\\": [
                    {
                        \\"Fn::Join\\": [
                            \\"\\",
                            [
                                \\"arn:\\",
                                {
                                    \\"Ref\\": \\"AWS::Partition\\"
                                },
                                \\":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole\\"
                            ]
                        ]
                    }
                ]
            }
        },
        \\"AWS679f53fac002430cb0da5b7982bd22872D164C4C\\": {
            \\"Type\\": \\"AWS::Lambda::Function\\",
            \\"Properties\\": {
                \\"Code\\": {
                    \\"S3Bucket\\": {
                        \\"Fn::Sub\\": \\"cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}\\"
                    },
                    \\"S3Key\\": \\"032a35d5de5db39968ca08635eb5461abf6a28bf3f742b27474151ede466fc69.zip\\"
                },
                \\"Role\\": {
                    \\"Fn::GetAtt\\": [
                        \\"AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2\\",
                        \\"Arn\\"
                    ]
                },
                \\"Handler\\": \\"index.handler\\",
                \\"Runtime\\": \\"nodejs12.x\\",
                \\"Timeout\\": 120
            },
            \\"DependsOn\\": [
                \\"AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2\\"
            ]
        }
    },
    \\"Parameters\\": {
        \\"BootstrapVersion\\": {
            \\"Type\\": \\"AWS::SSM::Parameter::Value<String>\\",
            \\"Default\\": \\"/cdk-bootstrap/hnb659fds/version\\",
            \\"Description\\": \\"Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store.\\"
        }
    },
    \\"Rules\\": {
        \\"CheckBootstrapVersion\\": {
            \\"Assertions\\": [
                {
                    \\"Assert\\": {
                        \\"Fn::Not\\": [
                            {
                                \\"Fn::Contains\\": [
                                    [
                                        \\"1\\",
                                        \\"2\\",
                                        \\"3\\",
                                        \\"4\\",
                                        \\"5\\"
                                    ],
                                    {
                                        \\"Ref\\": \\"BootstrapVersion\\"
                                    }
                                ]
                            }
                        ]
                    },
                    \\"AssertDescription\\": \\"CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.\\"
                }
            ]
        }
    }
}"
`;
